extensions/amp-analytics/0.1/events.js

```diff
@@ -1141,11 +1141,20 @@ export class VisibilityTracker extends EventTracker {
     const visibilitySpec = config['visibilitySpec'] || {};
     const selector = config['selector'] || visibilitySpec['selector'];
     const waitForSpec = visibilitySpec['waitFor'];
+    let reportWhenSpec = visibilitySpec['reportWhen'];
     const visibilityManager = this.root.getVisibilityManager();
     // special polyfill for eventType: 'hidden'
-    let createReadyReportPromiseFunc = null;
+    let createReportReadyPromiseFunc = null;
     if (eventType == 'hidden') {
-      createReadyReportPromiseFunc = this.createReportReadyPromise_.bind(this);
+      if (reportWhenSpec) {
```

Comment:
Since we already have the check fo `eventType == 'hidden'` here, I personally feel that another layer of  `createReportReadyPromise_ ` is not necessary. 

can we do
```
if (reportWhenSpec == 'hidden') {
  ....
  createReportReadyPromiseFunc = this. createReportReadyPromiseForHidden_.bind(this);
} else if (reportWhenSpec == 'endOfFrame') {
  createReportReadyPromiseFunc = this.createReportReadyPromiseForEndOfFrame_.bind(this);
}
...
```

Reply:
Done